src/AutoRest.CSharp/Common/Input/CodeModelConverter.cs

```diff
@@ -244,7 +244,8 @@ private InputModelType GetOrCreateModel(ObjectSchema schema, SchemaUsageProvider
                     : null,
                 DerivedModels: derived,
                 DiscriminatorValue: schema.DiscriminatorValue,
-                DiscriminatorPropertyName: schema.Discriminator?.Property.CSharpName());
+                DiscriminatorPropertyName: schema.Discriminator?.Property.CSharpName(),
```

Comment:
This is very likely to be wrong - we should use serialized name of the property because that will never change.

Reply:
Fixed in another PR